AntonBabunin/src/main/java/ua/artcode/market/controllers/ILoggingImpl.java

```diff
@@ -0,0 +1,92 @@
+package ua.artcode.market.controllers;
+
+import ua.artcode.market.interfaces.ILogging;
+import ua.artcode.market.models.Salesman;
+import ua.artcode.market.utils.Generator;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+public class ILoggingImpl implements ILogging {
+
+    private static volatile ILogging instance;
+
+    private List<Salesman> salesmenList;
+
+    private ILoggingImpl() throws IOException {
+        this.salesmenList = Generator.generateSalesmanList(0);
+        for (Salesman salesman : salesmenList){
+            write(String.format("Fullname: %s, Login: %s, Password: %s, \r\n",
+                    salesman.getFullName(), salesman.getLogin(),
+                    salesman.getPassword()));
+        }
+    }
+
+    public static ILogging getGetInstance() throws IOException {
+        if (instance == null) {
+            synchronized (ILoggingImpl.class) {
+                if (instance == null) {
+                    instance = new ILoggingImpl();
+                }
+            }
+        }
+        return instance;
+    }
+
+    @Override
+    public void write(String messege)
+            throws IOException {
+        File file = new File("C:\\Projects\\Java\\Gesserok\\aco22\\logs.txt");
+        FileWriter fileWriter = new FileWriter(file,true);
+        fileWriter.write(messege);
+        fileWriter.flush();
+        fileWriter.close();
+    }
+
+    @Override
+    public List<Salesman> getAllSalesmans() {
```

Comment:
why do you have this method in Logger class?

Reply:
Because it return a list of salesman, which are present in the database of sellers.